tcf_website/models/models.py

```diff
@@ -738,6 +738,20 @@ def filter_by_time(cls, days=None, start_time=None, end_time=None):
 
         query = query.filter(section_conditions)
         return query.distinct()
+    
+    @classmethod
+    def filter_by_gpa(cls, min_gpa=None):
```

Comment:
inline the filtering. performance is improved a bit if this the case. i know min_gpa may not be provided but it is trivial to work around this